File: problemas/1000/1094.py
from functools import reduce

n = int(input())

cobaias = {
    'C': 0,
    'R': 0,
    'S': 0
}

for i in range(n):
    linha = input().split(' ')
    cobaias[linha[1]] += int(linha[0])

total = reduce(lambda x, y: x + y, cobaias.values())
divisor = total / 100

print(f'Total: {total} cobaias')
print(f'Total de coelhos: {cobaias["C"]}')
print(f'Total de ratos: {cobaias["R"]}')
print(f'Total de sapos: {cobaias["S"]}')
print(f'Percentual de coelhos: {cobaias["C"] / divisor:.2f} %')
print(f'Percentual de ratos: {cobaias["R"] / divisor:.2f} %')
print(f'Percentual de sapos: {cobaias["S"] / divisor:.2f} %')

File: problemas/1000/1094.dart
import 'dart:io';

void main() {
  final n = int.parse(stdin.readLineSync());

  final cobaias = {
    'C': 0,
    'R': 0,
    'S': 0
  };

  for (var i = 0; i < n; i++) {
    final linha = stdin.readLineSync().split(' ');
    cobaias[linha[1]] += int.parse(linha[0]);
  }

  final total = cobaias.values.reduce((x, y) => x + y);
  final divisor = total / 100;

  stdout.writeln('Total: $total cobaias');
  stdout.writeln('Total de coelhos: ${cobaias['C']}');
  stdout.writeln('Total de ratos: ${cobaias['R']}');
  stdout.writeln('Total de sapos: ${cobaias['S']}');
  stdout.writeln('Percentual de coelhos: ${(cobaias['C'] / divisor).toStringAsFixed(2)} %');
  stdout.writeln('Percentual de ratos: ${(cobaias['R'] / divisor).toStringAsFixed(2)} %');
  stdout.writeln('Percentual de sapos: ${(cobaias['S'] / divisor).toStringAsFixed(2)} %');
}